Base/res/js/Spreadsheet/runtime.js

```diff
@@ -57,6 +57,56 @@ class Position {
         );
     }
 
+    range_up() {
+        if (this.row === 0) throw new Error(`No cells above this cell`);
+        const up_one = this.up(1);
+        let current_point = up_one;
+        for (
+            let point = current_point.up(1);
+            current_point.row !== 0 && typeof point.value() === "number";
```

Comment:
Why specifically a number and not _anything not empty_?

Reply:
why would it be helpful to have anything not number like?

Reply:
Not everything is a number, e.g. dates or (future) user-defined types

Reply:
This has been done by doing `val !== ''`, and can be iterated on in the future. :)